modules/tensor_mechanics/src/postprocessors/CriticalTimeStep.C

```diff
@@ -0,0 +1,81 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CriticalTimeStep.h"
+
+registerMooseObject("TensorMechanicsApp", CriticalTimeStep);
+
+template <>
+InputParameters
+validParams<CriticalTimeStep>()
+{
+  InputParameters params = validParams<ElementPostprocessor>();
+  params.addClassDescription(
+      "Computes and reports the critical time step for the explicit solver.");
+  params.addParam<MaterialPropertyName>(
+      "density",
+      "Name of Material Property  or a constant real number defining the density of the material.");
+  params.addParam<Real>("factor", 1.0, "Factor to mulitply to the critical time step.");
+  return params;
+}
+
+CriticalTimeStep::CriticalTimeStep(const InputParameters & parameters)
+  : ElementPostprocessor(parameters),
+    GuaranteeConsumer(this),
+    _material_density(getMaterialPropertyByName<Real>("density")),
+    _effective_stiffness(getMaterialPropertyByName<Real>("effective_stiffness")),
+    _factor(getParam<Real>("factor")),
+    _critical_time(parameters.isParamValid("critical_time"))
```

Comment:
What is this?  Why would you initialize a Real with a Boolean? Also I don't think critical_time is even a valid parameter name.